File: haskell/p-47.hs
and' = (&&)
or'  = (||)

nand a b = not (and' a b)
nor a b  = not (or' a b)
impl a b = or' (not a) b
equ' a b = a == b

table2 :: (Bool -> Bool -> Bool) -> IO ()
table2 f = mapM_ putStrLn [show a ++ " " ++ show b ++ " " ++ show (f a b)
                               | a <- [True, False], b <- [True, False]]

File: prolog/p-47.pl
:- ensure_loaded('p-46').

:- op(900, fy,not).
:- op(910, yfx, and).
:- op(910, yfx, nand).
:- op(920, yfx, or).
:- op(920, yfx, nor).
:- op(930, yfx, impl).
:- op(930, yfx, equ).
:- op(930, yfx, xor).



%% ?- table(A,B, A and (A or not B)).
%% true true true
%% true false true
%% false true false
%% false false false
%% false.
